### PR TITLE
Handle object result from the login()

### PR DESCRIPTION
**Pull request by netanelhugi**
* Added an option to return a result object from the login function.

### CODE REVIEW

File: lib/controller/handlers/callback.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import isObject from 'lodash/isObject'
```
Comment (berzniz) [inline lib/controller/handlers/callback.js:1]: Let's do the check without adding lodash as a dependency. It should be a simple test for us. For example test for boolean:
```
const isBoolean = (value === false || value === true)
```

Comment (berzniz) [reply lib/controller/handlers/callback.js:1]: or see:
image
https://youmightnotneed.com/lodash

We prefer to have no dependencies at all if possible for npm modules we publish

File: lib/controller/handlers/callback.js [hunk 2]
```diff
@@ -40,11 +41,26 @@ module.exports = {
...
             user,
-            login
+            login,
+            postResponseValidationErrorHandler
           }
         } = request
 
-        return Boolean(await login(request, user.nameID, user))
+        const loginResult = await login(request, user.nameID, user)
+        if (isObject(loginResult) && !loginResult.success) {
```
Comment (berzniz) [inline lib/controller/handlers/callback.js:50]: Unit tests are expected to be added to test all of these flows (see coverage is at 100%) for this one

File: lib/controller/handlers/callback.js (hunk 1 above)
Comment (netanelhugi) [reply lib/controller/handlers/callback.js]: Removed.

File: lib/controller/handlers/callback.js (hunk 2 above)
Comment (netanelhugi) [reply lib/controller/handlers/callback.js:51]: I added tests.

File: package.json [hunk 3]
```diff
@@ -16,7 +16,8 @@
     ]
   },
   "dependencies": {
-    "@node-saml/node-saml": "^4.0.3"
+    "@node-saml/node-saml": "^4.0.3",
+    "lodash": "^4.17.21"
```
Comment (berzniz) [inline package.json:20]: lodash -- remove

**Review by berzniz — APPROVED** [review]

**Pull request merged**